.github/workflows/mvn-verify-check.yml

```diff
@@ -30,7 +30,44 @@ env:
     -Dmaven.scaladoc.skip
 
 jobs:
+  cache-dependencies:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8']
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+      - run: |
+          set -x
+          max_retry=3; delay=30; i=1
+          while true; do
+            mvn help:evaluate -pl dist -Dexpression=included_buildvers \
+            -DforceStdout -PnoSnapshots -q | tr -d ',' | \
+            xargs -n 1 bash -c 'mvn initialize -pl sql-plugin-api -am -Dbuildver=$1' _ && break || {
+            if [[ $i -le $max_retry ]]; then
+              echo "mvn command failed. Retry $i/$max_retry."; ((i++)); sleep $delay; ((delay=delay*2))
+            else
+              echo "mvn command failed. Exit 1"; exit 1
+            fi
+          }
+          done 
+
+      - name: Get current date
+        run: |
+          echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ github.ref }}-${{ env.DATE }}
```

Comment:
This has to run before the L44, so that we can skip the job populating the cache if it is a cache hit

it would be great to have a separate step `gen-cache-key` that produces this whole key  with pom hash or target branch-yy.mm `${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-${{ env.DATE }}`
as an output.

Then you can reference this key on L67 and more importantly you can avoid repeating yourself and recomputing in all all downstream jobs by storing it as an output `daily-cache-key` of the `cache-dependencies` jobs. you can reference `needs.cache-dependencies.outputs.daily-cache-key`